e2e-tests/IIoTPlatform-E2E-Tests/deploy/IoTHubEdgeBaseDeployment.cs

```diff
@@ -0,0 +1,116 @@
+// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace IIoTPlatform_E2E_Tests.TestExtensions {
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+    using Microsoft.Azure.Devices;
+
+    /// <summary>
+    /// Default edge base deployment configuration
+    /// </summary>
+    public class IoTHubEdgeBaseDeployment : DeploymentConfiguration, IIoTHubDeployment {
+
+        /// <summary>
+        /// Create edge base deployer
+        /// </summary>
+        /// <param name="context"></param>
+        public IoTHubEdgeBaseDeployment(IIoTPlatformTestContext context) : base(context) {
+           _context = context ?? throw new ArgumentNullException(nameof(context));
+        }
+
+        /// <summary>
+        /// Create a new layered deployment or update an existing one
+        /// </summary>
+        public async Task<bool> CreateOrUpdateLayeredDeploymentAsync() {
+            var isSuccessful = false;
+            var configuration = await CreateOrUpdateConfigurationAsync(new Configuration(kDeploymentName) {
+                Content = new ConfigurationContent { ModulesContent = CreateDeploymentModules() },
+                TargetCondition = TestConstants.TargetCondition,
+                Priority = 0
+            }, true, kDeploymentName, new CancellationToken());
```

Comment:
Please see my comment above about `IIoTHubDeployment.CreateOrUpdateLayeredDeploymentAsync()` method. It should itself get a `CancellationToken` which it would thenpass down to `CreateOrUpdateConfigurationAsync()` method.

Reply:
changed to use cancellation token from interface